x-pack/plugin/search-business-rules/src/internalClusterTest/java/org/elasticsearch/xpack/searchbusinessrules/PinnedQueryBuilderIT.java

```diff
@@ -222,5 +233,42 @@ public void testHighlight() throws Exception {
         HighlightField highlight = highlights.get("field1");
         assertThat(highlight.fragments()[0].toString(), equalTo("<em>the</em> <em>quick</em> <em>brown</em> fox"));
     }
+
+    public void testMultiIndexDocs() throws Exception {
```

Comment:
It'd be great to add a test for the tricky case you uncovered around index aliases, where the same document could match multiple pinned entries.

Reply:
Added here: addc260d96204aaf481f0435e9426aadae679ff0